### PR TITLE
[RFC][UN-15030]Implement message system inside calendar

### PR DESCRIPTION
**Pull request by niceoutput**
https://uniplaces.atlassian.net/browse/UN-15030

screen shot 2018-11-23 at 14 52 37

### CODE REVIEW

File: tests/integration/components/uni-datepicker-test.js [hunk 1]
```diff
@@ -6,9 +6,45 @@ moduleForComponent('uni-datepicker', 'Integration | Component | uni datepicker',
...
+  ]);
 
-  this.render(hbs`{{uni-datepicker}}`);
+  this.render(hbs`{{uni-datepicker messages=messages}}`);
 
-  assert.notEqual(this.$().text().trim(), '');
+  assert.dom('p').hasText('This is a warning label 1');
+});
+
+test('it renders', function(assert) {
```
Comment (migueldemoura) [inline tests/integration/components/uni-datepicker-test.js]: Change test name to one that's more specific.

File: tests/integration/components/uni-datepicker-test.js [hunk 2]
```diff
@@ -6,9 +6,45 @@ moduleForComponent('uni-datepicker', 'Integration | Component | uni datepicker',
...
+    { type: 'information', label: 'This is a information label 2' },
+    { type: 'information', label: 'This is a information label 3' }
+  ]);
+
+  this.render(hbs`{{uni-datepicker messages=messages}}`);
+
+  assert.dom('p').hasText('This is a information label');
+});
+
+test('it renders', function(assert) {
```
Comment (migueldemoura) [inline tests/integration/components/uni-datepicker-test.js]: Change test name to one that's more specific.

File: tests/integration/components/uni-datepicker-test.js [hunk 3]
```diff
@@ -6,9 +6,45 @@ moduleForComponent('uni-datepicker', 'Integration | Component | uni datepicker',
...
+    { type: 'warning', label: 'This is a warning label 2' },
+    { type: 'warning', label: 'This is a warning label 3' }
+  ]);
+
+  this.render(hbs`{{uni-datepicker messages=messages}}`);
+
+  assert.dom('p').hasText('This is a warning label');
+});
+
+test('it renders', function(assert) {
```
Comment (migueldemoura) [inline tests/integration/components/uni-datepicker-test.js]: Change test name to one that's more specific.

File: tests/integration/components/uni-datepicker-test.js [hunk 4]
```diff
@@ -6,9 +6,45 @@ moduleForComponent('uni-datepicker', 'Integration | Component | uni datepicker',
 });
 
 test('it renders', function(assert) {
```
Comment (migueldemoura) [inline tests/integration/components/uni-datepicker-test.js]: Change test name to one that's more specific.

File: addon/components/uni-datepicker.js [hunk 5]
```diff
@@ -16,13 +16,25 @@ export default Component.extend({
...
+  selectedMessage: computed('messages', function() {
+    let messagesArray = this.get('messages');
+
+    return messagesArray.sort(function(a, b) {
+      if (a.type === b.type) {
+        return 0;
+      }
+
+      return a.type === 'information' ? 1 : -1;
+    })[0];
```
Comment (migueldemoura) [inline addon/components/uni-datepicker.js:42]: Do you want this to return undefined when the array is empty? I'd separate the sort from the arr[0] access too.

File: addon/components/uni-datepicker.js [hunk 6]
```diff
@@ -16,13 +16,25 @@ export default Component.extend({
...
   disabledDates: [],
+  messages: [],
 
   isFirstMonth: computed('center', function() {
     return this.get('center').isSame(this.get('minDate'), 'month') ? 'disabled' : '';
   }),
   isLastMonth: computed('center', function() {
     return this.get('center').isSame(this.get('maxDate'), 'month') ? 'disabled' : '';
   }),
+  selectedMessage: computed('messages', function() {
```
Comment (migueldemoura) [inline addon/components/uni-datepicker.js:29]: Can you add a comment on the expected behavior of this computed?

File: addon/components/uni-datepicker.js [hunk 7]
```diff
@@ -16,13 +16,25 @@ export default Component.extend({
...
+    let messagesArray = this.get('messages');
+
+    return messagesArray.sort(function(a, b) {
+      if (a.type === b.type) {
+        return 0;
+      }
+
+      return a.type === 'information' ? 1 : -1;
```
Comment (migueldemoura) [inline addon/components/uni-datepicker.js]: Move the message types to an enum

File: tests/integration/components/uni-datepicker-test.js [hunk 8]
```diff
@@ -6,9 +6,45 @@ moduleForComponent('uni-datepicker', 'Integration | Component | uni datepicker',
...
-  assert.expect(1);
+  this.set('messages', [
+    { type: 'information', label: 'This is a information label' },
```
Comment (migueldemoura) [inline tests/integration/components/uni-datepicker-test.js]: Use the enum types here, possibly even moving the messages to consts and reuse them across all tests.
Applies to all tests below.

**Review by migueldemoura — CHANGES_REQUESTED** [review]
Looking good :+1:

**Review by DaniFrancisco — CHANGES_REQUESTED** [review]
nothing to add to @migueldemoura 's comments

**Review by DaniFrancisco — APPROVED** [review]

**Review by sardyy — APPROVED** [review]
Nice job 👍

**Review by migueldemoura — APPROVED** [review]

**Pull request merged**